daemon-scheduler/pkg/store/environment_stm_applier_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package store
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/blox/blox/daemon-scheduler/pkg/json"
+	"github.com/blox/blox/daemon-scheduler/pkg/mocks"
+	"github.com/blox/blox/daemon-scheduler/pkg/types"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateApplierNoEnvironmentKey(t *testing.T) {
+	applier := &EnvironmentSTMApplier{}
+
+	err := applier.validateApplier()
+	assert.Error(t, err, "Expected error when record key is not set in applier")
+}
+
+func TestValidateApplierNoValidateAndUpdateFunction(t *testing.T) {
+	applier := &EnvironmentSTMApplier{
+		key: "someKey",
+	}
+
+	err := applier.validateApplier()
+	assert.Error(t, err, "Expected error when record key is not set in applier")
+}
+
+func TestUpdateEnvironment(t *testing.T) {
+	key := "key"
+
+	getEnv := environment(t)
+	getEnvJSON := environmentJSON(t, getEnv)
+
+	putEnv := environment(t)
+	putEnvJSON := environmentJSON(t, putEnv)
+
+	mockSTM := &mocks.MockSTM{
+		GetFunc: func(k string) string {
+			assert.Equal(t, key, k, "Unexpected key for Get")
+			return getEnvJSON
+		},
+		PutFunc: func(k string, val string, opts ...clientv3.OpOption) {
+			assert.Equal(t, key, k, "Unexpected key in Put")
+			assert.Equal(t, val, putEnvJSON, "Unexpected record in Put")
+		},
+	}
+
+	applier := &EnvironmentSTMApplier{
+		key: key,
+		validateAndUpdateEnv: func(exitingEnv *types.Environment) (*types.Environment, error) {
```

Comment:
nit, i think... should these four `exitingEnv` be `existingEnv`? I'm trying to figure out if this means that the existing environment is being passed in, or the exiting environment to be replaced. Normally i don't worry too much about typos, but I think it means two different things for `exiting` vs `existing`.